apps/roam/src/utils/conceptConversion.ts

```diff
@@ -0,0 +1,184 @@
+import { DiscourseNode } from "./getDiscourseNodes";
+import getDiscourseRelations from "./getDiscourseRelations";
+import type { DiscourseRelation } from "./getDiscourseRelations";
+import { Database } from "@repo/database/types.gen";
+import type { SupabaseContext } from "~/utils/supabaseContext";
+
+// When it's merged: import type { LocalConceptDataInput } from '@repo/database/input_types';
+type LocalConceptDataInput = Partial<Database["public"]["CompositeTypes"]["concept_local_input"]>;
+
+// Get data from roam about the node
+const getNodeExtraData = (node_uid: string): { author_uid: string, created: string, last_modified: string, page_uid: string } => {
+  const result = window.roamAlphaAPI.q(`[
+    :find
+      ?author_uid
+      ?page_uid
+      ?created
+      ?last_modified
+      :in $ ?block_uid
+      :where
+      [?block :block/uid ?block_uid]
+      [?block :create/user ?author_id]
+      [?author_id :user/uid ?author_uid]
+      [?block :create/time ?created]
+      [?block :edit/time ?last_modified]
+      [(get-else $ ?block :block/page ?block) ?page_id]
+      [?page_id :block/uid ?page_uid]
+  ]`, node_uid);
+  if (result.length != 1 || result[0].length != 4)
+    throw new Error("Invalid result from Roam query");
+
+  const [author_uid, page_uid, created_t, last_modified_t] = result[0] as [string, string, number, number];
+  const created = new Date(created_t).toISOString();
+  const last_modified = new Date(last_modified_t).toISOString();
+  return {
+    author_uid,
+    created,
+    last_modified,
+    page_uid
+  }
+}
+
+
+export function discourseNodeSchemaToLocalConcept(
+  context: SupabaseContext,
+  node: DiscourseNode, // contains type, uid, text
```

Comment:
Is this comment necessary?

Reply:
For someone unfamiliar with the codebase, yes. Will remove.